src/cpp/cpp_declarator_converter.cpp

```diff
@@ -69,8 +75,18 @@ symbolt &cpp_declarator_convertert::convert(
     scope=&cpp_typecheck.cpp_scopes.current_scope();
 
     // check the declarator-part of the type, in that scope
-    cpp_typecheck.typecheck_type(final_type);
+    //TODO: if it is a friend declaration we have to type-check it
```

Comment:
This comment doesn't seem to match the code? It seems like `if(!is_friend)` then we do the type-check in this scope(?) otherwise we defer to outside `cpp_save_scopet` ( @peterschrammel )